File: 剑指Offer/栈的压入弹出序列/isPopOrder.java
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
	
import java.util.ArrayList;
import java.util.Stack;
 
public class Solution {
    public boolean IsPopOrder(int [] pushA,int [] popA) {
        Stack<Integer> sta = new Stack<Integer>();
        int popIn = 0;
        for(int i = 0;i < pushA.length;i++){
            sta.push(pushA[i]);
            while(!sta.empty() && sta.peek() == popA[popIn]){
                sta.pop();
                popIn++;
            }
        }
        return sta.empty();
    }
}

File: 剑指Offer/栈的压入弹出序列/isPopOrder.py
# -*- coding:utf-8 -*-
class Solution:
    def IsPopOrder(self, pushV, popV):
        # write code here
        sta = []
        ind = 0
        for n in pushV:
            sta.append(n)
            while len(sta)>0 and sta[-1] == popV[ind]:
                sta.pop()
                ind+=1
        return len(sta)==0